### ISSUE TITLE
an error when batch size is 16

### ISSUE THREAD

**Issue by qianqing13579** [ROCmSoftwarePlatform/AMDMIGraphX#864]
when I  use resnet50 to do inference,and set batch size to 16,I find that the result of each image is different!

the model link: https://pan.baidu.com/s/1UcaJkzQoltul8MXOAMm_Mw
password：48fh

**Comment by qianqing13579** [comment]
@mvermeulen ,I have tested a lot of models by using MIGraphX1.2,and mobilenetV2 and efficientnet have the same problem. I think that maybe the group conv is the reason.

**Comment by scxiao** [comment]
@qianqing13579 for results are different, what do you mean for that? Can also can you provide an sample input and output for that? Thanks!

**Comment by qianqing13579** [comment]
@scxiao ,Here is a sample that show the problem.
link：https://pan.baidu.com/s/1gevFVPuQDxjwf3OecaxItQ 
password：262w 

The EfficientNet.tar includes the test code, test model and test image.The result of the sample is as follows:
==========0 image output=============
0.915527
0.084595
==========1 image output=============
0.947266
0.052856
the  results of two image in one batch are different!

**Comment by qianqing13579** [comment]
@scxiao ,here is a small model sample that show the problem.you can use it to find out the problem more easily
link：https://pan.baidu.com/s/1aqM0QKMI7sM89QSgd4i8dg 
password：tj30

I tried to solve the problem. I find the result of flatten op which is next to slice op is wrong. If I insert a  contiguous op between slice and flatten,the result will be right.I delete the eliminate_contiguous{"gpu::contiguous"} in the src/targets/gpu/target.cpp,and the result is also right.

**Comment by kahmed10** [comment]
@qianqing13579 can you please test #890 (branch `flatten_contiguous`) and see if it fixes resnet50 as well?

**Comment by qianqing13579** [comment]
@kahmed10 ，the mobilenet_v2 has the same problem. I have tried to solve the problem of mobilenet_v2. If I insert a contiguous op between broadcast and gpu::add, the result will be right. I think may be there is a logic problem in the eliminate_contiguous{"gpu::contiguous"} pass.

**Comment by kahmed10** [comment]
@qianqing13579 which build type does this error happen? I am not seeing on release build, but can see on debug. This may be a separate issue.

**Comment by qianqing13579** [comment]
@kahmed10 , I have used the lastest version of migraphx  to test mobilenet_v2 in release type. and I used FP16. The error happened again. If I insert a contiguous op between broadcast and gpu::add, the result will be right.

**Comment by qianqing13579** [comment]
@kahmed10 , I have used the lastest version of migraphx to test mobilenet_v2 in release type. and I used FP16. The error happened again. If I insert a contiguous op between broadcast and gpu::add, the result will be right.

**Comment by qianqing13579** [comment]
@kahmed10 ,here is my code:

#include <string>
#include <vector>
#include <migraphx/onnx.hpp>
#include <migraphx/gpu/target.hpp>
#include <migraphx/gpu/hip.hpp>
#include <migraphx/generate.hpp>
#include <migraphx/quantization.hpp>
#include <opencv2/opencv.hpp>
#include <opencv2/dnn.hpp>

using namespace cv;
using namespace cv::dnn;
using namespace std;
using namespace migraphx;

int main(int argc, char *argv[])
{
    // load
    migraphx::program net= migraphx::parse_onnx("./mobilenet_v2.onnx");
    std::pair<std::string, shape> inputAttribute=*(net.get_parameter_shapes().begin());
    string inputName=inputAttribute.first;
    shape inputShape=inputAttribute.second; 
    int N=inputShape.lens()[0];
    int C=inputShape.lens()[1];
    int H=inputShape.lens()[2];
    int W=inputShape.lens()[3];
    printf("input name:%s\n",inputName.c_str());
    printf("input shape:%d,%d,%d,%d\n",N,C,H,W);

    // compile
    migraphx::quantize_fp16(net);
    net.compile(migraphx::gpu::target{});

    // input
    int batchSize=N;
    migraphx::parameter_map m;
    for (std::pair<std::string, shape> x : net.get_parameter_shapes())
    {
        m[x.first] = migraphx::gpu::to_gpu(migraphx::generate_argument(x.second));
    }
    vector<Mat> srcImages;
    for(int i=0;i<batchSize;++i)
    {
        Mat srcImage=imread("../Test.jpg");
        srcImages.push_back(srcImage);
    }
    Mat inputBlob;
    blobFromImages(srcImages,inputBlob,1.0,cv::Size(W,H),cv::Scalar(0),false,false);
    m[inputName]= migraphx::gpu::to_gpu(migraphx::argument{inputShape, (float*)inputBlob.data}); 

    std::vector<argument> results = net.eval(m);
    argument result  = migraphx::gpu::from_gpu(results[0]);
    std::vector<float> logits;
    result.visit([&](auto output) { logits.assign(output.begin(), output.end()); });

    // output
    int numberOfClasses=logits.size()/N;
    for(int i=0;i<N;++i)
    {
        printf("==========%d image output=============\n",i);
        int startIndex=numberOfClasses*i;

        std::vector<float> logit;
        for(int j=0;j<numberOfClasses;++j)
        {
            printf("%f\n",logits[startIndex+j]);
        }
        
    }

    return 0;

}

**Comment by qianqing13579** [comment]
@kahmed10 ,I add a new issue about the problem.  #894

**Comment by causten** [comment]
Has the original problem been fix and now another issue been opened to address the next issue?   If so please close this issue.

**Comment by causten** [comment]
never mind I can close it
